src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -258,12 +258,45 @@ export class ExtHostTerminal extends Disposable {
 
 	public setInteractedWith(): boolean {
 		if (!this._state.isInteractedWith) {
-			this._state = { isInteractedWith: true };
+			this._state = {
+				...this._state,
+				isInteractedWith: true
+			};
 			return true;
 		}
 		return false;
 	}
 
+	public setShellType(shellType: TerminalShellType | number | undefined): boolean {
```

Comment:
We might be forced to do this conversion from string to number instead of TerminalShellType to number because shellType seems to be passed as string from the very beginning here: https://github.com/microsoft/vscode/blob/4add4d423d4d388e0530e6db0522f47210df0c7c/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1915 before we even derive our first onAnyInstance event..

Reply:
That does fire a `TerminalShellType`, the values are all just strings at runtime, is that the confusion?

Reply:
AHA! that is the confusion.

I actually did notice an unexpected behavior, where leaving the cases to use the types (such as PosixShellType.Sh) instead of 'sh' always casacded things down all the way to nu. So testing this code, I was always getting nu (which is the very last one in the switch case) no matter which shell I was using.

<img width="470" alt="Screenshot 2025-01-15 at 1 13 02 AM" src="https://github.com/user-attachments/assets/77c2b1d4-eeb3-43c3-ad09-157075e5adf1" />

As soon as I converted these into using string type as below:

<img width="537" alt="Screenshot 2025-01-15 at 1 15 20 AM" src="https://github.com/user-attachments/assets/24026f03-ea2a-40f1-b0cb-aa3f84b0b76f" />

everything started working fine..

Reply:
I have converted this back to using internal types, the reason why it didnt work was because I was missing break statements... lol